### DISCUSSION TITLE
Having Trouble Using anacrolix/log to Log to a File

### DISCUSSION THREAD

**Discussion by FIGBERT** [anacrolix/torrent#742] [Q&A] [answered]
Starting from the default logger, it's pretty simple to discard all the logs:
```go
config := torrent.NewDefaultClientConfig()
config.Logger = log.Default
config.Logger.Handlers = []log.Handler{log.DiscardHandler}
```

I can't quite get logging to a file to work, though:
```go
if file, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE, 0755); err == nil {
    handler := log.DefaultHandler
    handler.W = file
    config.Logger.Handlers = []log.Handler{handler}
}
```

**Comment by anacrolix** [comment]
Thanks I'll try this out and report back.

**Comment by FIGBERT** [comment, marked as answer]
Any updates?

**Reply by anacrolix** [reply]
Sorry for the delay, and the whacky log system. I was able to get it to work with a file, you still need to pass Debug=true in the `ClientConfig`, or set `GO_LOG=namespace=debug` (for example `*`, or `client` for how it's done in `cmd/torrent`, or some value that matches your logger's name).
```
diff --git a/cmd/torrent/download.go b/cmd/torrent/download.go
index 45164a3..ff25349 100644
--- a/cmd/torrent/download.go
+++ b/cmd/torrent/download.go
@@ -319,6 +319,13 @@ func downloadErr(flags downloadFlags) error {
 		if flags.Quiet {
 			logger = logger.FilterLevel(log.Critical)
 		}
+		file, err := os.OpenFile("client.log", os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
+		if err != nil {
+			return fmt.Errorf("opening client log: %w", err)
+		}
+		handler := log.DefaultHandler
+		handler.W = file
+		logger.Handlers = []log.Handler{handler}
 		clientConfig.Logger = logger
 	}
 	if flags.RequireFastExtension {
```

**Reply by FIGBERT** [reply]
Wonderful. Wasn't setting Debug to `true`, and I really need to learn more about how to use file permission in Go (with regards to `os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644`).

Thanks for the help, as always.
